Decred Wallet/Features/History/TransactionHistoryViewController.swift

```diff
@@ -55,140 +59,135 @@ class TransactionHistoryViewController: UIViewController {
+            print("wallet not synced on history")
             return
         }
-        self.visible = true
-        if (self.FromMenu){
-            refreshControl.showLoader(in: self.tableView)
-            loadTransactions()
-            FromMenu = true
-        }
+        self.refreshControl.showLoader(in: self.tableView)
+        loadTransactions()
     }
     
-    override func viewDidDisappear(_ animated: Bool) {
-        super.viewDidDisappear(animated)
-        self.visible = false
-        dismiss(animated: true, completion: nil)
+    func initFilterBtn() {
+        self.btnFilter.initMenu(self.filterMenu) { [weak self] index, value in
+            self?.applyTxFilter(currentFilter: DcrlibwalletTxFilterAll)
+        }
     }
     
-    @objc func handleRefresh(_ refreshControl: UIRefreshControl) {
-        self.loadTransactions()
+    func applyTxFilter(currentFilter: Int32) {
+        refreshControl.showLoader(in: self.tableView)
+        filteredItems.removeAll()
+        
+        self.filterActive = currentFilter != DcrlibwalletTxFilterAll
+        
+        switch currentFilter {
+        case DcrlibwalletTxDirectionSent:
+            self.filteredItems = self.transactions.filter{$0.Direction == DcrlibwalletTxDirectionSent && $0.Type == DcrlibwalletTxTypeRegular}
+            self.btnFilter.setTitle(LocalizedStrings.sent.appending("(\(self.filteredItems.count))"), for: .normal)
+            break
+        case DcrlibwalletTxDirectionReceived:
+            self.filteredItems = self.transactions.filter{$0.Direction == DcrlibwalletTxDirectionReceived && $0.Type == DcrlibwalletTxTypeRegular}
+            self.btnFilter.setTitle(LocalizedStrings.received.appending("(\(self.filteredItems.count))"), for: .normal)
+            break
+        case DcrlibwalletTxDirectionTransferred:
+            self.filteredItems = self.transactions.filter{$0.Direction == DcrlibwalletTxDirectionTransferred && $0.Type == DcrlibwalletTxTypeRegular}
+            self.btnFilter.setTitle(LocalizedStrings.yourself.appending("(\(self.filteredItems.count))"), for: .normal)
+            break
+        case DcrlibwalletTxFilterStaking:
+            self.filteredItems = self.transactions.filter{$0.Type == DcrlibwalletTxTypeRevocation || $0.Type == DcrlibwalletTxTypeTicketPurchase || $0.Type == DcrlibwalletTxTypeVote }
+            self.btnFilter.setTitle(LocalizedStrings.staking.appending("(\(self.filteredItems.count))"), for: .normal)
+            break
+        case DcrlibwalletTxFilterCoinBase:
+            self.filteredItems = self.transactions.filter{$0.Type == DcrlibwalletTxTypeCoinBase}
+            self.btnFilter.setTitle("COINBASE (\(self.filteredItems.count))", for: .normal)
+            break
+        default:
+            self.btnFilter.setTitle(LocalizedStrings.all.appending("(\(self.transactions.count))"), for: .normal)
+            break
+        }
+        self.refreshControl.endRefreshing()
     }
     
     func loadTransactions() {
```

Comment:
Should probably be `loadAllTransactions`

Reply:
changed in 9a8c24b